frontend/src/metabase/css/components/header.css

```diff
@@ -54,26 +54,34 @@
 .EditHeader .Button {
     color: var(--brand-color);
     border: none;
-    text-transform: uppercase;
-    font-size: 0.75rem;
-    background-color: rgba(255,255,255,0.5);
-    font-weight: normal;
+    font-size: 0.875rem;
+    background-color: rgba(255,255,255,0.1);
+    font-weight: 700;
     margin-left: 0.75em;
 }
 
 .EditHeader .Button--primary {
     background-color: white;
+    color: #444444;
 }
 
 .EditHeader .Button:hover {
     color: white;
     background-color: var(--brand-color);
 }
 
+.EditHeader.EditHeader--admin .Button:hover {
+  background-color: var(--admin-nav-bg-color);
+}
+
+.EditHeader.EditHeader--admin .Button.Button--primary:hover {
+  color: white;
```

Comment:
you can use `text-white-hover` on the edit-header buttons and we won't need this rule.